src/screens/WalletMigrationArchanova/SetAmount/WalletMigrationArchanovaSetAmount.js

```diff
@@ -18,94 +18,104 @@
     51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 import * as React from 'react';
+import { InteractionManager } from 'react-native';
 import { useNavigation } from 'react-navigation-hooks';
 import { useDispatch } from 'react-redux';
+import { BigNumber } from 'bignumber.js';
 import { useTranslationWithPrefix } from 'translations/translate';
 
 // Components
 import { Container, Content, Footer } from 'components/modern/Layout';
 import Button from 'components/modern/Button';
 import HeaderBlock from 'components/HeaderBlock';
-import ValueInput from 'components/ValueInput';
+import TokenValueInput from 'components/Inputs/TokenValueInput';
+import TokenBalanceAccessory from 'components/Inputs/TokenValueInput/TokenBalanceAccessory';
 
 // Constants
 import { CHAIN } from 'constants/chainConstants';
 
 // Selectors
-import { useRootSelector, useChainSupportedAssets, useChainRates, useFiatCurrency } from 'selectors';
-import { accountAssetsBalancesSelector } from 'selectors/balances';
+import { useSupportedAsset } from 'selectors/assets';
+import { useWalletAssetBalance } from 'selectors/balances';
 
 // Actions
 import { switchToArchanovaAccountIfNeededAction } from 'actions/accountsActions';
 import { setTokenToMigrateAction } from 'actions/walletMigrationArchanovaActions';
 
 // Utils
-import { findAssetByAddress, getAssetOption } from 'utils/assets';
-
-// Types
-import type { AssetOption } from 'models/Asset';
-
+import { wrapBigNumberOrNil } from 'utils/bigNumber';
+import { isValidValueForTransfer, isLessThanOrEqualToBalance } from 'utils/transactions';
 
 function WalletMigrationArchanovaSetAmount() {
   const { t, tRoot } = useTranslationWithPrefix('walletMigrationArchanova.setAmount');
   const navigation = useNavigation();
   const dispatch = useDispatch();
 
+  const inputRef = React.useRef();
+
   const address: string = navigation.getParam('address');
-  const initialBalance: string = navigation.getParam('balance');
+  const initialValue = wrapBigNumberOrNil(navigation.getParam('value'));
 
-  const assetOption = useAssetOption(address);
+  const asset = useSupportedAsset(CHAIN.ETHEREUM, address);
+  const balance = useWalletAssetBalance(CHAIN.ETHEREUM, address);
 
-  const [balance, setBalance] = React.useState(initialBalance ?? '');
-  const [isValid, setIsValid] = React.useState(true);
+  const [value, setValue] = React.useState<?BigNumber>(initialValue);
 
   // Force active account to be archanova
   React.useEffect(() => {
     dispatch(switchToArchanovaAccountIfNeededAction());
   }, [dispatch]);
 
+  React.useEffect(() => {
+    InteractionManager.runAfterInteractions(() => inputRef.current?.focus());
+  }, []);
```

Comment:
```suggestion
  React.useEffect(() => {
    const handler = InteractionManager.runAfterInteractions(() => inputRef.current?.focus());
    return () => { InteractionManager.clearInteractionHandle(handler); };
  }, []);
```

Reply:
Fixed in a slightly different manner, as `clearInteractionHandle` is for `createInteractionHandle`.